### ISSUE TITLE
Rule proposal: disallow defaults in defineModel

### ISSUE THREAD

**Issue by spaansba** [vuejs/eslint-plugin-vue#3137]
**Please describe what the rule should do:**

It could be useful to have a small opt-in rule that disallows setting a `default` value in `defineModel()`.

Vue documents that a default can cause the parent and child values to become de-synchronized when the parent does not provide a value. Some projects may therefore prefer to avoid defaults on models entirely and initialize the value in the parent instead.

A possible name could be `vue/no-define-model-default`, but I do not have a strong preference on the name or whether this fits better as an option on an existing rule.

**What category should the rule belong to?**

[ ] Enforces code style (layout)
[X] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
<script setup lang="ts">
const modelValue = defineModel<string>({ default: '' })
</script>
```

```vue
<script setup lang="ts">
const count = defineModel<number>('count', { default: 0 })
</script>
```

```vue
<script setup lang="ts">
const items = defineModel<string[]>({ default: () => [] })
</script>
```

Calls without a default would remain valid:

```vue
<script setup lang="ts">
const modelValue = defineModel<string>()
const count = defineModel<number>('count')
</script>
```

**Additional context**

Vue documentation warns about possible parent/child de-synchronization when `defineModel()` has a default:

https://vuejs.org/guide/components/v-model.html#under-the-hood
